topics/imaging/tutorials/tutorial-CP/tutorial.md

```diff
@@ -0,0 +1,633 @@
+> ### {% icon hands_on %} Hands-on: Specify metadata to CellProfiler
+>
+> 1. **Starting Modules** {% icon tool %} with the following parameters:
+>    - Images
+>       - *"Do you want to filter only the images? "*: `Select the images only`
+>    - Metadata
+>       - *"Do you want to extract the metadata?"*: `Yes, specify metadata`
+>           - *"Metadata extraction method"*: `Extract from file/folder names`
+>       - *"Metadata source"*: `File name`
+>       - *"Select the pattern to extract metadata from the file name"*: `field1__field2__field3__field4__field5__field6`
+>       - *"Extract metadata from"*: `All images`
+>    - NamesAndTypes
+>       - *"Process 3D"*: `No, do not process 3D data` 
+>       - *"Assign a name to"*: `Give every image the same name` 
+>       - *"Name to assign these images"*: `DNA` 
+>       - *"Select the image type"*: `Grayscale image` 
+>           - *"Set intensity range from"*: `Image metadata` 
+>    - Groups
+>       - *"Do you want to group your images?"*: `Yes, group the images` 
+>       - *"param"*: `field1` 
+>
+>
+>
+>    > ### {% icon comment %} Comment
+>    >
+>    > The images downloaded from the IDR are named following the pattern: `plate__imageID__cropX__cropY__cropWidth__cropHeight`. These fields indicate to which plate the image belongs, what is the identifier of the image in the IDR, and the 4 cropping parameters selected. In our case, the upper-left corner (X, Y) and the width and height from there. We have a total of 6 metadata values encoded in the name of the file, separated by `__`. The pattern to extract our metadata from the file name properly is, therefore: `field1__field2__field3__field4__field5__field6`. It is important to keep in mind that, later in the analysis, our `plate` will be `field1`, `imageID` will be `field2`, `cropX` will be `field3`, etc. for CellProfiler.
+>    {: .comment}
+>
+{: .hands_on}
+
+# Object segmentation
+
+## Segment nuclei
+
+Since we are interested in segmenting the nucleoli, you may wonder why we need to segment nuclei first. There are several reasons for that: 
+
+- Get the nuclei features. The intensity, size, shape, number of nucleoli per nucleus, etc. can be informative to study the nucleoli.
+
+- Avoid the detection of wrong spots. In a first segmentation, we detect the nuclei, while in the second pass we will segment the holes (nucleoli). The holes need to fall inside the nuclei, and hence the importance of having them segmented too.
+
+In the first step, we will identify the nuclei that are complete, meaning that they are not touching the borders of the image.
+
+> ### {% icon hands_on %} Hands-on: Segment nuclei that are complete within the boundaries of the image
+>
+> 1. **IdentifyPrimaryObjects** {% icon tool %} with the following parameters:
+>    - {% icon param-file %} *"Select the input CellProfiler pipeline"*: `output_pipeline` (output of **Starting Modules** {% icon tool %})
```

Comment:
I was initially confused with this name `output_pipeline` because the output of the preceding tool was not named `output_pipeline` (silly me!). I don't know what is the best practice in Galaxy: renaming outputs every time we run a tool? Or keeping the output name (here would be `3: Starting Modules`).

Reply:
I don't know the best practices either. Each tool is giving a pipeline as output so I guess it makes sense that we keep that name but I see that it can be confusing...